solutions/usaco-1063.mdx

```diff
@@ -0,0 +1,97 @@
+---
+id: usaco-1063
+source: USACO Silver 2020 December
+title: Rectangular Pasture
+author: Kevin Sheng
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_prob2_silver_dec20.html)
+
+## Video Solution
+
+<Youtube id="AH1wyxq8nPM" />
+
+## Implementation
+
+Note: the algorithm in the code may differ from that described in the video.
+
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class RPasture {
+	public static void main(String[] args) throws IOException {
+		BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
+		int cowNum = Integer.parseInt(read.readLine());
+
+		HashSet<Integer> seenX = new HashSet<>();
+		HashSet<Integer> seenY = new HashSet<>();
+		int[][] cows = new int[cowNum][2];
+		for (int c = 0; c < cowNum; c++) {
+			cows[c] = Arrays.stream(read.readLine().split(" "))
+					.mapToInt(Integer::parseInt).toArray();
+
+			seenX.add(cows[c][0]);
+			seenY.add(cows[c][1]);
+		}
+
+		/*
+		 * because all the xs and ys are distinct,
+		 * we can just sort of "compress" them into just 1, 2, 3, etc..
```

Comment:
coordinate compression is explained [here](https://usaco.guide/silver/sorting-custom?lang=java#coordinate-compression), so you can (optionally) add a link?

Reply:
sure i can do that